guidance/README.adoc

```diff
@@ -0,0 +1,31 @@
+= Guiding principles for Automation Good Practices
+
+As an overall guiding principle for designing good automation, the Zen of Python, by Tim Peters, serves as a good guidepost to follow. The principles defined in this are very applicable and can give guidance when the better practice is  unclear.
+
+----
+The Zen of Python, by Tim Peters
+
+Beautiful is better than ugly.
+Explicit is better than implicit.
+Simple is better than complex.
+Complex is better than complicated.
+Flat is better than nested.
+Sparse is better than dense.
+Readability counts.
+Special cases aren't special enough to break the rules.
+Although practicality beats purity.
+Errors should never pass silently.
+Unless explicitly silenced.
+In the face of ambiguity, refuse the temptation to guess.
+There should be one-- and preferably only one --obvious way to do it.
+Although that way may not be obvious at first unless you're Dutch.
+Now is better than never.
+Although never is often better than *right* now.
+If the implementation is hard to explain, it's a bad idea.
+If the implementation is easy to explain, it may be a good idea.
+Namespaces are one honking great idea -- let's do more of those!
+----
+
+Examples::
+Some example situations that can be pulled out from this include the use of 'ignore_errors'. Since Errors should neve pass silently, a better practice would be using conditionals and skipping irrelevant tasks.
```

Comment:
"Since Errors should neve pass silently"

should be:

"Since Errors should never pass silently"

Reply:
Also, since you bring up ignore_errors, I would say a good use of ignore_errors could also be to register the error and use a fail task intentionally with a more useful debug message.

Reply:
added additional notes and fixed the typo

Reply:
this one is approved from 2022-03-09